language/move-vm/runtime/src/loader.rs

```diff
@@ -999,6 +1011,18 @@ impl Loader {
             })
             .map_err(expect_no_verification_errors)?;
 
+        fail::fail_point!("verifier-failpoint-2", |_| { Ok(module.clone()) });
+
+        if self.runtime_config.paranoid_type_checks {
+            if &module.self_id() != id {
```

Comment:
Maybe we should always check this, not dependent on `paranoid_type_checks`, in case the storage return the wrong module.